src/EnergyPlus/WaterUse.cc

```diff
@@ -1694,5 +1695,18 @@ namespace WaterUse {
                 (state.dataHeatBal->Zone(ZoneNum).Multiplier * state.dataHeatBal->Zone(ZoneNum).ListMultiplier);
         }
     }
+
+    Real64 calcH2ODensity(EnergyPlusData &state)
+    {
+        static constexpr std::string_view RoutineName{"calcH2ODensity"};
+
+        if (state.dataWaterUse->calcRhoH2O) {
+            int DummyValue = 1;
+            state.dataWaterUse->rhoH2OStd = FluidProperties::GetDensityGlycol(state, "WATER", Constant::InitConvTemp, DummyValue, RoutineName);
+            state.dataWaterUse->calcRhoH2O = false;
+        }
+        return state.dataWaterUse->rhoH2OStd;
+    }
```

Comment:
Hmmm.  I'm inclined to push back slightly on this and see if we can do a little better.  @RKStrand, what happens if you just put the function call right in the state class constructor, and avoid this extra flag.  The changes would be, stream of consciousness:

- At the bottom of WaterUse.hh, in the state definition, keep the new `Real64 rhoH2OStd;` variable definition
- But add an empty constructor declaration there: `WaterUseData::WaterUseData();`
- Then here in WaterUse.cc, actually implement the constructor, so something like:

```c++
WaterUseData::WaterUseData() {
  this->->rhoH2OStd = FluidProperties::GetDensityGlycol(state, "WATER", Constant::InitConvTemp, DummyValue, RoutineName);
}
```

That way it will only be calculated once, when the whole state is constructed.  And you don't need to add the bool flag anymore.  And it can still be dynamic if the water properties change.  What do you think?  As I said, this is stream of consciousness, so maybe I'm missing something, but it seems to be a cleaner way to 'one-time init' a variable.

Reply:
I tried this real quick myself and the problem with it is that the GetDensityGlycol function isn't callable without state, so this idea can just be ignored.  This can carry on as-is...  sorry for the distraction @RKStrand

Reply:
@Myoldmopar Ok--that sounds good.  As @rraustad has reviewed it and given it a thumbs up, what else needs to happen here other than get it up to speed with develop and then merge?